### PR TITLE
Make tidy work on stylo trees

### LINKED ISSUE #14855: Make tidy fall back to the default os.walk if git isn't available
Right now, if we run `./mach test-tidy` it relies on `git` to get the changed files from the last merge. This will result in an error if we're not in a git repo (like stylo, which is based on mercurial). Tidy should check whether git is available, and if it's not (`subprocess.check_output('git status')` will throw a `fatal: ...` in that case), `only_changed_files` should be set to `False`.

Code: `python/tidy/servo_tidy/tidy.py`

### PR DESCRIPTION
**Pull request by Manishearth**
This adds `./mach test-tidy --stylo`, which turns off the usage of git to search for files to check (fixes #14855), and also turns off the wpt lint, making it possible to run tidy in a stylo tree.

r? @jdm @wafflespeanut

cc @bzbarsky

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14853)

### CODE REVIEW

**Comment by Manishearth** [comment]
An alternate fix would be to provide a different config file for stylo and severely reduce the checks being run

**Comment by Wafflespeanut** [comment]
I'm not really comfortable with changing all those functions to be stylo-specific. We should be making tidy more generic by moving servo-specific stuff out of it. I have a suggestion. What if we have support for per-directory config in tidy? So, initially there's the root config which applies to all the files/dirs in tree. In addition to that, the root config can specify config overrides for some of the subdirs. Those overrides could be in a separate config file in those subdirs. Tidy will apply that config for the subdirs.

File: python/tidy/servo_tidy/tidy.py [hunk 1]
```diff
@@ -991,25 +1001,27 @@ def check_commits(path='.'):
     raise StopIteration
 
 
-def scan(only_changed_files=False, progress=True):
+def scan(only_changed_files=False, progress=True, stylo=False):
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: We should override `only_changed_files` to `False` if we have `stylo`. That way, some of the functions (`FileList` and `filter_files`) don't need another argument.

File: python/tidy/servo_tidy/tidy.py [hunk 2]
```diff
@@ -947,7 +957,7 @@ def check(self):
         module = imp.load_source(filename[:-3], self.path)
         if hasattr(module, 'Lint'):
             if issubclass(module.Lint, LintRunner):
-                lint = module.Lint(self.path, self.only_changed_files, self.exclude_dirs, self.progress)
+                lint = module.Lint(self.path, self.only_changed_files, self.exclude_dirs, self.progress, stylo=self.stylo)
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: I don't like this, but I don't really have a great idea here. So be it :smile:

File: python/tidy/servo_tidy/tidy.py [hunk 3]
```diff
@@ -991,25 +1001,27 @@ def check_commits(path='.'):
...
     # other lint checks
-    lint_errors = run_lint_scripts(only_changed_files, progress)
+    lint_errors = run_lint_scripts(only_changed_files, progress, stylo=stylo)
     # check commits for WIP
-    commit_errors = check_commits()
+    commit_errors = []
+    if not stylo:
+       commit_errors = check_commits()
     # chain all the iterators
-    errors = itertools.chain(config_errors, directory_errors, file_errors, dep_license_errors, lint_errors,
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Why were these removed?

File: python/tidy/servo_tidy/tidy.py [hunk 4]
```diff
@@ -991,25 +1001,27 @@ def check_commits(path='.'):
...
     # check dependecy licenses
     dep_license_errors = check_dep_license_errors(get_dep_toml_files(only_changed_files), progress)
     # other lint checks
-    lint_errors = run_lint_scripts(only_changed_files, progress)
+    lint_errors = run_lint_scripts(only_changed_files, progress, stylo=stylo)
     # check commits for WIP
-    commit_errors = check_commits()
+    commit_errors = []
+    if not stylo:
+       commit_errors = check_commits()
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Python foo :stuck_out_tongue: 

``` python
commit_errors = [] if stylo else check_commits()
```

**Review by ghost — COMMENTED** [review]
Okay, we'll land this.

**Comment by jdm** [comment]
What's happening here?

File: python/tidy/servo_tidy/tidy.py (hunk 1 above)
Comment (Manishearth) [inline python/tidy/servo_tidy/tidy.py]: They still need two different arguments, you want `./mach test-tidy --all` to work.

**Comment by Manishearth** [comment]
Updated. r?

File: python/tidy/servo_tidy/tidy.py (hunk 1 above)
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: I'm saying that we should override it inside the `scan` function when we have `stylo`. `FileList` iterator should bail out when it doesn't find git, regardless of whether we're in stylo or not.

**Comment by Wafflespeanut** [comment]
Tidy fails:

    ./python/tidy/servo_tidy/tidy.py:1071: E241 multiple spaces after ','
    ./python/tidy/servo_tidy/tidy.py:1009: Line is longer than 120 characters

**Comment by jdm** [comment]
@Manishearth Want to finish this off by fixing the tidy errors?

**Comment by Wafflespeanut** [comment]
Seriously?

**Comment by Manishearth** [comment]
Updated. r? @jdm

File: python/tidy/servo_tidy/tidy.py (hunk 1 above)
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Never mind. I'll get it done with an easy issue :smile:
